lib/matplotlib/colorbar.py

```diff
@@ -1565,32 +1565,55 @@ def make_axes_gridspec(parent, *, location=None, orientation=None,
     location = kw['ticklocation'] = loc_settings['location']
 
     pad = loc_settings["pad"]
-    wh_space = 2 * pad / (1 - pad)
-
-    # for shrinking
-    pad_s = (1 - shrink) * 0.5
-    wh_ratios = [pad_s, shrink, pad_s]
 
     # we need to none the tree of layoutboxes because constrained_layout can't
     # remove and replace the tree hierarchy w/o a segfault.
     layoutbox.nonetree(parent.get_subplotspec().get_gridspec()._layoutbox)
     if location == "left":
+        wh_space = 2 * pad / (1 - pad)
+        anchor = kw.pop('anchor', (0.0, 0.5))
+        panchor = kw.pop('panchor', (1.0, 0.5))
+
+        # for shrinking
+        wh_ratios = [(1-anchor[1])*(1-shrink), shrink, anchor[1]*(1-shrink)]
+
         gs = parent.get_subplotspec().subgridspec(
             1, 2, wspace=wh_space, width_ratios=[fraction, 1-fraction-pad])
         ss_main = gs[1]
         ss_cb = gs[0].subgridspec(3, 1, hspace=0, height_ratios=wh_ratios)[1]
     elif location == "right":
+        wh_space = 2 * pad / (1 - pad)
```

Comment:
This code is exactly the same for left and right?  Should it be, and if so, can it be factored out?

Reply:
It is correct. I will rewrite the code and factor out the duplicate code.